candidategenerator.js

```diff
@@ -167,12 +167,42 @@ window.GEN = function(sTree, words, options){
 	return candidates;
 }
 
-function rootIsHeaded(pRoot, recCat) {
-	var children = pRoot.children || [];
-	for (var i = 0; i < children.length; i++)
-		if (children[i].cat === pCat.nextLower(pRoot.cat))
+/* Function to check if a tree obeys headedness. Each node must either be be
+ * terminal or have at least one child of the category immidately below its own
+ * on the prosodic hierarch. Otherwise, return false. Written as a recursive
+ * function, basically a constraint.
+ */
+function obeysHeadedness(tree){
+	//inner function
+	function nodeIsHeaded(node) {
+		/* Function to check if a node is headed. Relies on the prosodic hierarchy being
+		 * properly defined. Returns true iff one of the node's children is of the
+		 * category directly below its own category on the prosodic hierarchy or the
+		 * node is terminal.
+		 */
+		var children = node.children;
+		//vacuously true if node is terminal
+		if (!children)
 			return true;
-	return false;
+		for (var i = 0; i < children.length; i++)
+			if (children[i].cat === pCat.nextLower(node.cat))
+				return true;
+			return false;
```

Comment:
Please add the curly braces for the for loop and if statement; right now the indentation  makes it look like "return false" is inside the loop, which it (correctly) isn't really.